src/schema/sources.ts

```diff
@@ -419,13 +536,13 @@ const validateSquadData = ({
 export const ensureSourcePermissions = async (
   ctx: Context,
   sourceId: string | undefined,
-  permission = SourcePermissions.View,
+  props?: Partial<SourcePermissionsProps>,
```

Comment:
Instead of merging both props into one variable, I suggest separating them as in most cases the permission will be provided and the validate against will not. So this will make it a bit shorter to call.

Reply:
It was optional previously and wanted to keep supporting it but we can definitely make this required and update existing implementations of the omitted third parameter.